src/lib.rs

```diff
@@ -29,12 +29,15 @@
 //! This crate uses the same syntax for format strings as the [strftime()]
 //! (http://man7.org/linux/man-pages/man3/strftime.3.html) function from the C
 //! standard library.
-
 #![doc(html_logo_url = "https://www.rust-lang.org/logos/rust-logo-128x128-blk-v2.png",
        html_favicon_url = "https://www.rust-lang.org/favicon.ico",
        html_root_url = "https://doc.rust-lang.org/time/")]
 #![allow(trivial_numeric_casts)]
+#![allow(unused_imports)]
 #![cfg_attr(test, deny(warnings))]
+#![cfg_attr(test, allow(deprecated))]
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
Oh this is baked into that derive mode... If that's the case then this can be conditional, but the library should have an explicit `extern crate core` and use imports from there rather than conditionally importing from core/std